File: OpenRobertaServer/src/test/resources/crossCompilerTests/_expected/robotSpecific/targetLanguage/ev3lejosv0/sensors_default.java
package generated.main;

import de.fhg.iais.roberta.runtime.*;
import de.fhg.iais.roberta.runtime.ev3.*;

import de.fhg.iais.roberta.mode.general.*;
import de.fhg.iais.roberta.mode.action.*;
import de.fhg.iais.roberta.mode.sensor.*;
import de.fhg.iais.roberta.mode.action.ev3.*;
import de.fhg.iais.roberta.mode.sensor.ev3.*;

import de.fhg.iais.roberta.components.*;

import java.util.LinkedHashSet;
import java.util.HashMap;
import java.util.Set;
import java.util.Map;
import java.util.List;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;

import lejos.remote.nxt.NXTConnection;

public class NEPOprog {
    private static Configuration brickConfiguration;

    private Set<UsedSensor> usedSensors = new LinkedHashSet<UsedSensor>(Arrays.asList(new UsedSensor(SensorPort.S1, SensorType.TOUCH, TouchSensorMode.TOUCH), new UsedSensor(SensorPort.S4, SensorType.ULTRASONIC, UltrasonicSensorMode.DISTANCE), new UsedSensor(SensorPort.S4, SensorType.ULTRASONIC, UltrasonicSensorMode.PRESENCE), new UsedSensor(SensorPort.S3, SensorType.COLOR, ColorSensorMode.COLOUR), new UsedSensor(SensorPort.S3, SensorType.COLOR, ColorSensorMode.RED), new UsedSensor(SensorPort.S3, SensorType.COLOR, ColorSensorMode.AMBIENTLIGHT), new UsedSensor(SensorPort.S3, SensorType.COLOR, ColorSensorMode.RGB), new UsedSensor(SensorPort.S2, SensorType.GYRO, GyroSensorMode.ANGLE), new UsedSensor(SensorPort.S2, SensorType.GYRO, GyroSensorMode.RATE)));
    private Hal hal = new Hal(brickConfiguration, usedSensors);

    private void ____sensors() {
        hal.drawText(String.valueOf(hal.isPressed(SensorPort.S1)), ___numberVar, ___numberVar);
        hal.drawText(String.valueOf(hal.getUltraSonicSensorDistance(SensorPort.S4)), ___numberVar, ___numberVar);
        hal.drawText(String.valueOf(hal.getUltraSonicSensorPresence(SensorPort.S4)), ___numberVar, ___numberVar);
        hal.drawText(String.valueOf(hal.getColorSensorColour(SensorPort.S3)), ___numberVar, ___numberVar);
        hal.drawText(String.valueOf(hal.getColorSensorRed(SensorPort.S3)), ___numberVar, ___numberVar);
        hal.drawText(String.valueOf(hal.getColorSensorAmbient(SensorPort.S3)), ___numberVar, ___numberVar);
        hal.drawText(String.valueOf(hal.getColorSensorRgb(SensorPort.S3)), ___numberVar, ___numberVar);
        hal.resetRegulatedMotorTacho(ActorPort.B);
        hal.drawText(String.valueOf(hal.getRegulatedMotorTachoValue(ActorPort.B, MotorTachoMode.DEGREE)), ___numberVar, ___numberVar);
        hal.drawText(String.valueOf(hal.getRegulatedMotorTachoValue(ActorPort.B, MotorTachoMode.ROTATION)), ___numberVar, ___numberVar);
        hal.drawText(String.valueOf(hal.getRegulatedMotorTachoValue(ActorPort.B, MotorTachoMode.DISTANCE)), ___numberVar, ___numberVar);
        hal.drawText(String.valueOf(hal.isPressed(BrickKey.ENTER)), ___numberVar, ___numberVar);
        hal.drawText(String.valueOf(hal.isPressed(BrickKey.UP)), ___numberVar, ___numberVar);
        hal.drawText(String.valueOf(hal.isPressed(BrickKey.DOWN)), ___numberVar, ___numberVar);
        hal.drawText(String.valueOf(hal.isPressed(BrickKey.LEFT)), ___numberVar, ___numberVar);
        hal.drawText(String.valueOf(hal.isPressed(BrickKey.RIGHT)), ___numberVar, ___numberVar);
        hal.drawText(String.valueOf(hal.isPressed(BrickKey.ESCAPE)), ___numberVar, ___numberVar);
        hal.drawText(String.valueOf(hal.isPressed(BrickKey.ANY)), ___numberVar, ___numberVar);
        hal.resetGyroSensor(SensorPort.S2);
        hal.drawText(String.valueOf(hal.getGyroSensorAngle(SensorPort.S2)), ___numberVar, ___numberVar);
        hal.drawText(String.valueOf(hal.getGyroSensorRate(SensorPort.S2)), ___numberVar, ___numberVar);
        hal.drawText(String.valueOf(hal.getTimerValue(1)), ___numberVar, ___numberVar);
        hal.drawText(String.valueOf(hal.getTimerValue(2)), ___numberVar, ___numberVar);
        hal.drawText(String.valueOf(hal.getTimerValue(3)), ___numberVar, ___numberVar);
        hal.drawText(String.valueOf(hal.getTimerValue(4)), ___numberVar, ___numberVar);
        hal.drawText(String.valueOf(hal.getTimerValue(5)), ___numberVar, ___numberVar);
        hal.resetTimer(1);
        hal.resetTimer(2);
        hal.resetTimer(3);
        hal.resetTimer(4);
        hal.resetTimer(5);
    }

    private void ____waitUntil() {
        while ( true ) {
            if ( hal.isPressed(SensorPort.S1) == true ) {
                break;
            }
            hal.waitFor(15);
        }
        while ( true ) {
            if ( hal.getUltraSonicSensorDistance(SensorPort.S4) < 30 ) {
                break;
            }
            hal.waitFor(15);
        }
        while ( true ) {
            if ( hal.getUltraSonicSensorPresence(SensorPort.S4) == true ) {
                break;
            }
            hal.waitFor(15);
        }
        while ( true ) {
            if ( hal.getColorSensorColour(SensorPort.S3) == PickColor.RED ) {
                break;
            }
            hal.waitFor(15);
        }
        while ( true ) {
            if ( hal.getColorSensorRed(SensorPort.S3) < 50 ) {
                break;
            }
            hal.waitFor(15);
        }
        while ( true ) {
            if ( hal.getColorSensorAmbient(SensorPort.S3) < 50 ) {
                break;
            }
            hal.waitFor(15);
        }
        while ( true ) {
            if ( hal.getRegulatedMotorTachoValue(ActorPort.B, MotorTachoMode.DEGREE) > 180 ) {
                break;
            }
            hal.waitFor(15);
        }
        while ( true ) {
            if ( hal.getRegulatedMotorTachoValue(ActorPort.B, MotorTachoMode.ROTATION) > 2 ) {
                break;
            }
            hal.waitFor(15);
        }
        while ( true ) {
            if ( hal.getRegulatedMotorTachoValue(ActorPort.B, MotorTachoMode.DISTANCE) < 30 ) {
                break;
            }
            hal.waitFor(15);
        }
        while ( true ) {
            if ( hal.isPressed(BrickKey.ENTER) == true ) {
                break;
            }
            hal.waitFor(15);
        }
        while ( true ) {
            if ( hal.isPressed(BrickKey.UP) == true ) {
                break;
            }
            hal.waitFor(15);
        }
        while ( true ) {
            if ( hal.isPressed(BrickKey.DOWN) == true ) {
                break;
            }
            hal.waitFor(15);
        }
        while ( true ) {
            if ( hal.isPressed(BrickKey.LEFT) == true ) {
                break;
            }
            hal.waitFor(15);
        }
        while ( true ) {
            if ( hal.isPressed(BrickKey.RIGHT) == true ) {
                break;
            }
            hal.waitFor(15);
        }
        while ( true ) {
            if ( hal.isPressed(BrickKey.ESCAPE) == true ) {
                break;
            }
            hal.waitFor(15);
        }
        while ( true ) {
            if ( hal.isPressed(BrickKey.ANY) == true ) {
                break;
            }
            hal.waitFor(15);
        }
        while ( true ) {
            if ( hal.getTimerValue(1) > 500 ) {
                break;
            }
            hal.waitFor(15);
        }
        while ( true ) {
            if ( hal.getTimerValue(2) > 500 ) {
                break;
            }
            hal.waitFor(15);
        }
        while ( true ) {
            if ( hal.getTimerValue(3) > 500 ) {
                break;
            }
            hal.waitFor(15);
        }
        while ( true ) {
            if ( hal.getTimerValue(4) > 500 ) {
                break;
            }
            hal.waitFor(15);
        }
        while ( true ) {
            if ( hal.getTimerValue(5) > 500 ) {
                break;
            }
            hal.waitFor(15);
        }
    }

    public static void main(String[] args) {
        try {
             brickConfiguration = new EV3Configuration.Builder()
                .setWheelDiameter(5.6)
                .setTrackWidth(18.0)
                .addActor(ActorPort.B, new Actor(ActorType.LARGE, true, DriveDirection.FOREWARD, MotorSide.RIGHT))
                .addSensor(SensorPort.S1, new Sensor(SensorType.TOUCH))
                .addSensor(SensorPort.S2, new Sensor(SensorType.GYRO))
                .addSensor(SensorPort.S3, new Sensor(SensorType.COLOR))
                .addSensor(SensorPort.S4, new Sensor(SensorType.ULTRASONIC))
                .build();

            new NEPOprog().run();
        } catch ( Exception e ) {
            Hal.displayExceptionWaitForKeyPress(e);
        }
    }

    float ___numberVar = 0;
    boolean ___booleanVar = true;
    String ___stringVar = "";
    PickColor ___colourVar = PickColor.WHITE;
    NXTConnection ___connectionVar = null;
    ArrayList<Float> ___numberList = new ArrayList<>(Arrays.asList((float) 0, (float) 0));
    ArrayList<Boolean> ___booleanList = new ArrayList<>(Arrays.<Boolean>asList(true, true));
    ArrayList<String> ___stringList = new ArrayList<>(Arrays.<String>asList("", ""));
    ArrayList<PickColor> ___colourList = new ArrayList<>(Arrays.<PickColor>asList(PickColor.WHITE, PickColor.WHITE));
    ArrayList<NXTConnection> ___connectionList = new ArrayList<>(Arrays.<NXTConnection>asList(___connectionVar, ___connectionVar));

    public void run() throws Exception {
        hal.startLogging();
        ____sensors();
        hal.closeResources();

    }
}

File: OpenRobertaServer/src/test/resources/crossCompilerTests/_expected/robotSpecific/targetLanguage/ev3dev/sensors_default.py
#!/usr/bin/python

from __future__ import absolute_import
from roberta.ev3 import Hal
from ev3dev import ev3 as ev3dev
import math
import os
import time

class BreakOutOfALoop(Exception): pass
class ContinueLoop(Exception): pass

_brickConfiguration = {
    'wheel-diameter': 5.6,
    'track-width': 18.0,
    'actors': {
        'B':Hal.makeLargeMotor(ev3dev.OUTPUT_B, 'on', 'foreward'),
    },
    'sensors': {
        '1':Hal.makeTouchSensor(ev3dev.INPUT_1),
        '2':Hal.makeGyroSensor(ev3dev.INPUT_2),
        '3':Hal.makeColorSensor(ev3dev.INPUT_3),
        '4':Hal.makeUltrasonicSensor(ev3dev.INPUT_4),
    },
}
hal = Hal(_brickConfiguration)

___numberVar = 0
___booleanVar = True
___stringVar = ""
___colourVar = 'white'
___connectionVar = None
___numberList = [0, 0]
___booleanList = [True, True]
___stringList = ["", ""]
___colourList = ['white', 'white']
___connectionList = [___connectionVar, ___connectionVar]
def ____sensors():
    global ___numberVar, ___booleanVar, ___stringVar, ___colourVar, ___connectionVar, ___numberList, ___booleanList, ___stringList, ___colourList, ___connectionList
    hal.drawText(str(hal.isPressed('1')), ___numberVar, ___numberVar)
    hal.drawText(str(hal.getUltraSonicSensorDistance('4')), ___numberVar, ___numberVar)
    hal.drawText(str(hal.getUltraSonicSensorPresence('4')), ___numberVar, ___numberVar)
    hal.drawText(str(hal.getColorSensorColour('3')), ___numberVar, ___numberVar)
    hal.drawText(str(hal.getColorSensorRed('3')), ___numberVar, ___numberVar)
    hal.drawText(str(hal.getColorSensorAmbient('3')), ___numberVar, ___numberVar)
    hal.drawText(str(hal.getColorSensorRgb('3')), ___numberVar, ___numberVar)
    hal.resetMotorTacho('B')
    hal.drawText(str(hal.getMotorTachoValue('B', 'degree')), ___numberVar, ___numberVar)
    hal.drawText(str(hal.getMotorTachoValue('B', 'rotation')), ___numberVar, ___numberVar)
    hal.drawText(str(hal.getMotorTachoValue('B', 'distance')), ___numberVar, ___numberVar)
    hal.drawText(str(hal.isKeyPressed('enter')), ___numberVar, ___numberVar)
    hal.drawText(str(hal.isKeyPressed('up')), ___numberVar, ___numberVar)
    hal.drawText(str(hal.isKeyPressed('down')), ___numberVar, ___numberVar)
    hal.drawText(str(hal.isKeyPressed('left')), ___numberVar, ___numberVar)
    hal.drawText(str(hal.isKeyPressed('right')), ___numberVar, ___numberVar)
    hal.drawText(str(hal.isKeyPressed('escape')), ___numberVar, ___numberVar)
    hal.drawText(str(hal.isKeyPressed('any')), ___numberVar, ___numberVar)
    hal.resetGyroSensor('2')
    hal.drawText(str(hal.getGyroSensorValue('2', 'angle')), ___numberVar, ___numberVar)
    hal.drawText(str(hal.getGyroSensorValue('2', 'rate')), ___numberVar, ___numberVar)
    hal.drawText(str(hal.getTimerValue(1)), ___numberVar, ___numberVar)
    hal.drawText(str(hal.getTimerValue(2)), ___numberVar, ___numberVar)
    hal.drawText(str(hal.getTimerValue(3)), ___numberVar, ___numberVar)
    hal.drawText(str(hal.getTimerValue(4)), ___numberVar, ___numberVar)
    hal.drawText(str(hal.getTimerValue(5)), ___numberVar, ___numberVar)
    hal.resetTimer(1)
    hal.resetTimer(2)
    hal.resetTimer(3)
    hal.resetTimer(4)
    hal.resetTimer(5)

def ____waitUntil():
    global ___numberVar, ___booleanVar, ___stringVar, ___colourVar, ___connectionVar, ___numberList, ___booleanList, ___stringList, ___colourList, ___connectionList
    while True:
        if hal.isPressed('1') == True:
            break
        hal.waitFor(15)
    while True:
        if hal.getUltraSonicSensorDistance('4') < 30:
            break
        hal.waitFor(15)
    while True:
        if hal.getUltraSonicSensorPresence('4') == True:
            break
        hal.waitFor(15)
    while True:
        if hal.getColorSensorColour('3') == 'red':
            break
        hal.waitFor(15)
    while True:
        if hal.getColorSensorRed('3') < 50:
            break
        hal.waitFor(15)
    while True:
        if hal.getColorSensorAmbient('3') < 50:
            break
        hal.waitFor(15)
    while True:
        if hal.getMotorTachoValue('B', 'degree') > 180:
            break
        hal.waitFor(15)
    while True:
        if hal.getMotorTachoValue('B', 'rotation') > 2:
            break
        hal.waitFor(15)
    while True:
        if hal.getMotorTachoValue('B', 'distance') < 30:
            break
        hal.waitFor(15)
    while True:
        if hal.isKeyPressed('enter') == True:
            break
        hal.waitFor(15)
    while True:
        if hal.isKeyPressed('up') == True:
            break
        hal.waitFor(15)
    while True:
        if hal.isKeyPressed('down') == True:
            break
        hal.waitFor(15)
    while True:
        if hal.isKeyPressed('left') == True:
            break
        hal.waitFor(15)
    while True:
        if hal.isKeyPressed('right') == True:
            break
        hal.waitFor(15)
    while True:
        if hal.isKeyPressed('escape') == True:
            break
        hal.waitFor(15)
    while True:
        if hal.isKeyPressed('any') == True:
            break
        hal.waitFor(15)
    while True:
        if hal.getTimerValue(1) > 500:
            break
        hal.waitFor(15)
    while True:
        if hal.getTimerValue(2) > 500:
            break
        hal.waitFor(15)
    while True:
        if hal.getTimerValue(3) > 500:
            break
        hal.waitFor(15)
    while True:
        if hal.getTimerValue(4) > 500:
            break
        hal.waitFor(15)
    while True:
        if hal.getTimerValue(5) > 500:
            break
        hal.waitFor(15)

def run():
    global ___numberVar, ___booleanVar, ___stringVar, ___colourVar, ___connectionVar, ___numberList, ___booleanList, ___stringList, ___colourList, ___connectionList
    ____sensors()

def main():
    try:
        run()
    except Exception as e:
        hal.drawText('Fehler im EV3', 0, 0)
        hal.drawText(e.__class__.__name__, 0, 1)
        hal.drawText(str(e), 0, 2)
        hal.drawText('Press any key', 0, 4)
        while not hal.isKeyPressed('any'): hal.waitFor(500)
        raise

if __name__ == "__main__":
    main()